modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx

```diff
@@ -34,35 +54,85 @@ export default function EditObjectFolder({
 }: EditObjectFolder) {
 	const [
 		{
+			baseResourceURL,
+			deleteObjectDefinition,
+			editObjectDefinitionURL,
 			elements,
 			isLoadingObjectFolder,
+			modelBuilderModals,
 			objectDefinitionsStorageTypes,
 			objectFolderName,
 			rightSidebarType,
+			selectedObjectDefinitionNode,
 			selectedObjectFolder,
 			showChangesSaved,
 		},
 		dispatch,
 	] = useObjectFolderContext();
 
+	const [
+		objectRelationshipParameterRequired,
+		setObjectRelationshipParameterRequired,
+	] = useState(false);
+
 	const store = useStore();
 
 	const {nodes} = store.getState();
 
-	const [showModal, setShowModal] = useState<ModelBuilderModals>({
-		addObjectDefinition: false,
-		addObjectField: false,
-		addObjectFolder: false,
-		addObjectRelationship: false,
-		deleteObjectDefinition: false,
-		deleteObjectFolder: false,
-		deleteObjectRelationship: false,
-		editObjectDefinitionExternalReferenceCode: false,
-		editObjectFolder: false,
-		moveObjectDefinition: false,
-		publishObjectDefinitions: false,
-		redirectToEditObjectDefinitionDetails: false,
-	});
+	const handleDeleteObjectDefinition = (
+		deleteObjectDefinition: DeletedObjectDefinition
+	) => {
+		dispatch({
+			payload: {
+				newDeleteObjectDefinition: deleteObjectDefinition,
+			},
+			type: TYPES.SET_DELETE_OBJECT_DEFINITION,
+		});
+	};
+
+	const updateModelBuilderStructure = async (
+		newObjectRelationshipId: number
+	) => {
+		const payload = await getUpdatedModelBuilderStructurePayload(
+			selectedObjectFolder.name
+		);
+
+		dispatch({
+			payload: {
+				...payload,
+				rightSidebarType: 'objectRelationshipDetails',
+				selectedObjectRelationshipId: newObjectRelationshipId,
+			},
+			type: TYPES.UPDATE_MODEL_BUILDER_STRUCTURE,
+		});
+
+		dispatch({
+			payload: {
+				selectedObjectRelationshipId: newObjectRelationshipId,
+			},
+			type: TYPES.SET_SELECTED_OBJECT_RELATIONSHIP_EDGE,
+		});
+	};
+
+	useEffect(() => {
+		const makeFetch = async () => {
+			if (selectedObjectDefinitionNode) {
+				const url = createResourceURL(baseResourceURL, {
+					objectDefinitionId: selectedObjectDefinitionNode?.data?.id,
```

Comment:
Hey Arthur, the point of adding this validation of the selectedobjectdefinitionNode was to avoid using so many optional chaining operators. So you can use:

`selectedObjectDefinitionNode.data.id,`

Reply:
done!